tests/unit/test.config.js

```diff
@@ -0,0 +1,185 @@
+import path from 'path';
+
+import {assert} from 'chai';
+import {describe, it} from 'mocha';
+import sinon from 'sinon';
+import {fs} from 'mz';
+
+import {Program} from '../../src/program';
+import {
+  applyConfigToArgv,
+  parseConfig,
+  loadJSConfigFile,
+} from '../../src/config';
+import {withTempDir} from '../../src/util/temp-dir';
+import {UsageError} from '../../src/errors';
+
+function makeArgv({
+  userCmd = [],
+  command = 'fakecommand',
+  commandDesc = 'this is a fake command',
+  commandExecutor = sinon.stub(),
+  commandOpt,
+  globalOpt,
+}) {
+  const program = new Program(userCmd);
+
+  if (globalOpt) {
+    program.setGlobalOptions(globalOpt);
+  }
+  if (commandOpt) {
+    program.command(command, commandDesc, commandExecutor, commandOpt);
+  }
+  return {
+    argv: program.yargs.exitProcess(false).argv,
+    defaultValues: program.defaultValues,
+  };
+}
+
+describe('config', () => {
+  describe('applyConfigToArgv', () => {
+    it('overrides the default value with a configured value', () => {
```

Comment:
This test isn't doing what it says because the option is not configured with a default value. You already added 'preserves configured value over default' so I think you can delete this one.